feature/policy_forwarding/encapsulation/otg_tests/decap_gre_ipv4/decap_gre_ipv4_test.go

```diff
@@ -0,0 +1,836 @@
+
+	// Configuring Static Route: IPV6-DST2 --> ATE Port 2
+	configStaticRoute(t, dut, IPv6Dst2, otgPort2.IPv6)
+
+	// Policy Based Forwading Rule-1
+	// configureGRETunnel(t, dut)
+
+	// Test cases.
+	type testCase struct {
+		Name        string
+		Description string
+		testFunc    func(t *testing.T, dut *ondatra.DUTDevice, otgConfig *otg.OTG, config gosnappi.Config)
+	}
+
+	testCases := []testCase{
+		{
+			Name:        "Testcase-GreDecapIPv4Traffic",
+			Description: "GRE Decapsulation of IPv4 traffic",
+			testFunc:    testGreDecapIPv4,
+		},
+		{
+			Name:        "Testcase-GreDecapIPv6Traffic",
+		 	Description: "GRE Decapsulation of IPv6 traffic",
+		 	testFunc:    testGreDecapIPv6,
+		},
+		{
+			Name:        "Testcase-GreDecapIPv4MPLSTraffic",
+			Description: "GRE Decapsulation of IPv4-over-MPLS traffic",
+			testFunc:    testGreDecapIPv4MPLS,
+		},
+		{
+		 	Name:        "Testcase-GreDecapIPv6MPLSTraffic",
+		 	Description: "GRE Decapsulation of IPv6-over-MPLS traffic",
+		 	testFunc:    testGreDecapIPv6MPLS,
+		},
+		{
+			Name:        "Testcase-GreDecapMultiLabelMPLSTraffic",
+		 	Description: "GRE Decapsulation of multi-label MPLS traffic",
+		 	testFunc:    testGreDecapMultiLabelMPLS,
+		},
+		{
+			Name:        "Testcase-GrePassthroughTraffic",
+			Description: "GRE Pass-through (Negative)",
+		 	testFunc:    testGrePassthrough,
+		},
+	}
+
+	// Run the test cases.
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Logf("Description: %s", tc.Description)
+			tc.testFunc(t, dut, otgConfig, config)
+		})
+	}
+
+}
+
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
+	p1 := dut.Port(t, "port1")
+	gnmi.Replace(t, dut, d.Interface(p1.Name()).Config(), configInterfaceDUT(p1, dutPort1, dut))
+	p2 := dut.Port(t, "port2")
+	gnmi.Replace(t, dut, d.Interface(p2.Name()).Config(), configInterfaceDUT(p2, dutPort2, dut))
+
+	// Configure Network instance type on DUT
+	t.Log("Configure/update Network Instance")
+	fptest.ConfigureDefaultNetworkInstance(t, dut)
+
+	if deviations.ExplicitInterfaceInDefaultVRF(dut) {
+		fptest.AssignToNetworkInstance(t, dut, p1.Name(), deviations.DefaultNetworkInstance(dut), 0)
+		fptest.AssignToNetworkInstance(t, dut, p2.Name(), deviations.DefaultNetworkInstance(dut), 0)
+	}
+
+	if deviations.ExplicitPortSpeed(dut) {
+		fptest.SetPortSpeed(t, p1)
+		fptest.SetPortSpeed(t, p2)
+	}
+}
+
+// Configures the given DUT interface.
+func configInterfaceDUT(p *ondatra.Port, a *attrs.Attributes, dut *ondatra.DUTDevice) *oc.Interface {
+	i := a.NewOCInterface(p.Name(), dut)
+	s4 := i.GetOrCreateSubinterface(0).GetOrCreateIpv4()
+	if deviations.InterfaceEnabled(dut) && !deviations.IPv4MissingEnabled(dut) {
+		s4.Enabled = ygot.Bool(true)
+	}
+	i.GetOrCreateSubinterface(0).GetOrCreateIpv6()
+
+	return i
+}
+
+// Congigure Static Routes on DUT
+func configStaticRoute(t *testing.T, dut *ondatra.DUTDevice, prefix string, nexthop string) {
```

Comment:
Should use 
https://github.com/openconfig/featureprofiles/blob/5bd6746949359f4a3851e735c9773b040909b65a/internal/cfgplugins/staticroute.go#L38 instead

Reply:
Added the changes